### PR TITLE
SNC-103 added POC test-runner implementation to internals

### PR DESCRIPTION
**Pull request by davidmdm**
Need a test runner!

### CODE REVIEW

File: cpa/tester/result.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	"github.com/CircleCI-Public/circle-policy-agent/cpa"
+	"github.com/CircleCI-Public/circle-policy-agent/internal"
+	"gopkg.in/yaml.v3"
+)
+
+type Result struct {
+	Group string
+	Name  string
+	Ok    bool
+	Err   error
```
Comment (sagar-connect) [inline cpa/tester/result.go:20]: error is more like message, for example "no rules" isn't really an error.

File: cpa/tester/result.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+	"github.com/CircleCI-Public/circle-policy-agent/cpa"
+	"github.com/CircleCI-Public/circle-policy-agent/internal"
+	"gopkg.in/yaml.v3"
+)
+
+type Result struct {
+	Group string
+	Name  string
+	Ok    bool
```
Comment (sagar-connect) [inline cpa/tester/result.go:19]: should be called `passed`?

**Review by sagar-connect — APPROVED** [review]

**Review by gosuku — APPROVED** [review]

File: cpa/tester/result.go (hunk 1 above)
Comment (davidmdm) [reply cpa/tester/result.go:20]: I wanted to change the type to string at some point, but then I can't easily test for Error types, ErrNoPolicies, ErrNoTests, or future known errors, and then make an informed decision about whether the test is passing or not.

File: cpa/tester/result.go (hunk 2 above)
Comment (davidmdm) [reply cpa/tester/result.go:19]: Sure that's fine by me. Will change.

**Review by rooneyshuman — APPROVED** [review]

**Pull request merged**